tests/JsonReaderTests.cs

```diff
@@ -766,6 +766,56 @@ public void String_AsEnum_Doesnt_Move_Reader()
         TestMovesReaderPastReadValue(JsonReader.String().AsEnum<JsonValueKind>(), /*lang=json*/ @"""Null""");
     }
 
+    [Theory]
+    [InlineData("foobar", /*lang=json*/ @"""foobar""")]
+    [InlineData(null, /*lang=json*/ "null")]
+    public void String_Nullable_With_Valid_Input(string? expected, string json)
+    {
+        var reader = JsonReader.String().Nullable(null);
+        var result = reader.Read(json);
+
+        Assert.Equal(expected, result);
+    }
+
+    [Fact]
```

Comment:
This should test that all of the following are also invalid:

- `true`
- `false`
- a JSON number
- a JSON array
- a JSON object

Reply:
Increased the coverage for invalid input with 84c70c2042d581bbd8945c9d7f09bc111eea4906